app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -714,60 +859,102 @@ public void setTitleToUrl(int serviceId, String videoUrl, String name) {
                 }
             }
         }
-    }
+    }*/
 
     @Override
     public boolean onBackPressed() {
         if (DEBUG) Log.d(TAG, "onBackPressed() called");
+
+        // If we are in fullscreen mode just exit from it via first back press
+        if (player != null && player.isInFullscreen()) {
+            if (!PlayerHelper.isTablet(activity)) player.onPause();
+            restoreDefaultOrientation();
+            setAutoplay(false);
+            return true;
+        }
+
+        // If we have something in history of played items we replay it here
+        boolean isPreviousCanBePlayed = player != null
+                && player.getPlayQueue() != null
+                && player.videoPlayerSelected()
+                && player.getPlayQueue().previous();
+        if (isPreviousCanBePlayed) {
+            return true;
+        }
```

Comment:
No need to create a variable
```suggestion
        if (player != null
                && player.getPlayQueue() != null
                && player.videoPlayerSelected()
                && player.getPlayQueue().previous()) {
            // If we have something in history of played items we replay it here
            return true;
        }
        
```